src/python/pants/init/extension_loader.py

```diff
@@ -27,30 +30,31 @@ class PluginLoadOrderError(PluginLoadingError):
     pass
 
 
+def _requirement_key(req: Requirement) -> str:
```

Comment:
I feel like there is a requirement name normalization function somewhere that we already use?

Reply:
Ah yes, `canonicalize_name` in `packaging.utils`. Shouldn't we be deferring to that?

Reply:
Yes, we should. Will fix.